backend/src/main/java/fk/prof/backend/model/policy/ZookeeperBasedPolicyStoreAPI.java

```diff
@@ -0,0 +1,232 @@
+                    inMemoryPolicyCache.put(decode32(appId), decode32(clusterId), decode32(procName), policyDetails);
+                }
+            }
+        }
+    }
+
+    void init() throws Exception {
+        synchronized (this) {
+            if (!initialized) {
+                populateCacheFromZK();
+                initialized = true;
+            }
+        }
+    }
+
+    Set<String> getAppIds(String prefix) {
+        if (prefix == null) return new HashSet<>();
+        Set<String> appIds = inMemoryPolicyCache.getAppId();
+        return appIds.stream().filter(s -> s.startsWith(prefix)).collect(Collectors.toSet());
+
+    }
+
+    Set<String> getClusterIds(String appId, String prefix) {
+        if (prefix == null) return new HashSet<>();
+        Set<String> clusterIds = inMemoryPolicyCache.getClusterIds(appId);
+        return clusterIds.stream().filter(s -> s.startsWith(prefix)).collect(Collectors.toSet());
+    }
+
+    Set<String> getProcNames(String appId, String clusterId, String prefix) {
+        if (prefix == null) return new HashSet<>();
+        Set<String> procNames = inMemoryPolicyCache.getProcNames(appId, clusterId);
+        return procNames.stream().filter(s -> s.startsWith(prefix)).collect(Collectors.toSet());
+    }
+
+    @Override
+    public PolicyDTO.PolicyDetails getPolicy(Recorder.ProcessGroup processGroup) {
+        if (processGroup == null) {
+            logger.info("Process Group is null");
+            return null;
+        }
+        return inMemoryPolicyCache.get(processGroup.getAppId(), processGroup.getCluster(), processGroup.getProcName());
+    }
+
+    @Override
+    public Future<Void> createPolicy(Recorder.ProcessGroup processGroup, PolicyDTO.PolicyDetails policyDetails) {
+        Future<Void> future = Future.future();
+        if (processGroup == null || policyDetails == null) {
+            future.fail("One of processGroup(" + processGroup + ") or policyDetails(" + policyDetails + ") is null");
+            return future;
+        }
+        if (getPolicy(processGroup) != null) {
+            future.fail("Policy already exists");
+            return future;
+        }
+        setPolicy(future, processGroup, policyDetails);
+        return future;
+    }
+
+    @Override
+    public Future<Void> updatePolicy(Recorder.ProcessGroup processGroup, PolicyDTO.PolicyDetails policyDetails) {
+        Future<Void> future = Future.future();
+        if (processGroup == null || policyDetails == null) {
+            future.fail("One of processGroup(" + processGroup + ") or policyDetails(" + policyDetails + ") is null");
+            return future;
+        }
+        if (getPolicy(processGroup) == null) {
+            future.fail("Policy does not exists ");
+            return future;
+        }
+        if (getPolicy(processGroup).equals(policyDetails)) {
```

Comment:
Is this check required? If you add it here, you will have to add this check in UI as well. We can simply choose to not care about this scenario unless you see a strong reason why we should